src/components/dialog/use-dialog-appearance.ts

```diff
@@ -0,0 +1,31 @@
+import { RefObject, useEffect, useRef } from 'react'
+
+interface UseDialogAppearanceOptions {
```

Comment:
**quibble:** I try to use `Options` for things that are explicitly optional (I clearly didn't do that in my POC 😅) and instead use `Input` when we're dealing with a param that needs to be supplied; i.e. `UseDialogAppearanceInput`. Not a big deal though and very subjective!

Reply:
**suggestion:** might be worth if all of us have an agreement on how we declare the naming convention for the interfaces